flocker/node/agents/ebs.py

```diff
@@ -440,8 +441,11 @@ def _next_device(self, instance_id):
             instance are currently occupied.
         """
         volumes = self.connection.get_all_volumes()
-        devices = [v.attach_data.device for v in volumes
-                   if v.attach_data.instance_id == instance_id]
+        devices = pset({v.attach_data.device for v in volumes
+                       if v.attach_data.instance_id == instance_id})
+        devices = devices.add(devices_in_use)
```

Comment:
This is wrong - you're adding the set as an *item*. So now you have `pset(["/dev/sdf", pset(["/dev/sdg"])])`.

You want `devices = devices | devices_in_use`.

Given the logic is broken I think this merits more thought on testing, I'll try to address that later in review.